### ISSUE TITLE
Add choice of more icons in Warp

### ISSUE THREAD

**Issue by dannyneira** [warpdotdev/Warp#5408]
### Dupe Check

- [X] I have [searched Warp feature requests](https://github.com/warpdotdev/warp/issues?q=is%3Aissue+label%3A%22Feature+requests%22) and there are no duplicates

### Describe the solution you'd like?

Inspired by comment from a user on the Warp Icon feature request. https://github.com/warpdotdev/Warp/issues/2703#issuecomment-2423424346

> I do appreciate the work that goes into designing a new icon, so no hate on a totally subjective thing like this. In fact, it would be great if users had the choice of choosing more icons from the Warp team or elsewhere. Do you think it is feasible to do build icons into appearance/theme settings? Arc browser has a good example of this.

Arc Browser Icon example:

![CleanShot 2024-10-23 at 08 58 05](https://github.com/user-attachments/assets/7386327a-f2be-4f58-8f24-06efd9b9387e)

### Is your feature request related to a problem? Please describe.

I can't customize the icon or choose the original or new icons for Warp.

### Additional context

To anyone else interested in this feature, please add a 👍 to the original post at the top to signal that you want this feature, and subscribe if you'd like to be notified.

### How important is this feature to you?

3

### Warp Internal (ignore) - linear-label:39cc6478-1249-4ee7-950b-c428edfeecd1

None

**Comment by dannyneira** [comment]
To anyone else interested in this feature, please add a 👍 to the original post at the top to signal that you want this feature, and subscribe if you'd like to be notified.

**Comment by Z2Flow** [comment]
The new icon violates Einstein's dictum on simplicity:

`Everything should be made as simple as possible, but not simpler`

**Comment by scosman** [comment]
Warp and Cursor are a constant source of confusion for me.

<img width="120" alt="Screenshot 2024-11-07 at 8 56 23 AM" src="https://github.com/user-attachments/assets/4406b774-1284-4ec1-bdf3-8b48eb3ad1ee">

I've even made a script to update the Warp icon, but that ends up invalidating the app's signature and breaking gatekeeper and automatic updates.

**Comment by berkus** [comment]
Please include this file in the selection :)

I am tired sick of replacing it manually after each upgrade...

[classic_1984_mac.icns.zip](https://github.com/user-attachments/files/17995945/classic_1984_mac.icns.zip)

**Comment by pquerner** [comment]
Nice! Do you have plans for Linux support as well?

**Comment by scosman** [comment]
@dannyneira Super glad to see this! 

Hopefully there's one "terminal" look icon in there? Terminal.app, iterm, ubuntu terminal, and countless others have taught me to look for a "> _" or "$ _" icon.

**Comment by berkus** [comment]
Seriously, reach out to author of https://github.com/dhanishgajjar/terminal-icons - he'll be happy to be credited with such great terminal, you'll be happy to have high-quality icons.

**Comment by neilcitruslime** [comment]
Please implement this feature I find the new icon hard to locate visually in the Dock it looks like other icons now.

**Comment by berkus** [comment]
I've added a little article on how to replace app icon automatically for those who want to install their own custom: [check it out](https://metta.systems/blog/replace-app-icon-automatically/). Merry Xmas!

**Comment by moluuser** [comment]
I found that these docs show how to change the dock icon on Warp's settings. But it has not appeared on the latest app.
https://docs.warp.dev/appearance/custom-dock-icons

**Comment by dannyneira** [comment]
@moluuser Thanks for your patience. It turns out this is going to take some additional work to implement correctly (we found some bugs). We'd dedicate more time to getting it right and past the finish line. We'll post further updates on this thread. The page is still available with the direct link, but hidden in the nav menu for the docs since the feature isn't live yet.

**Comment by dannyneira** [comment]
Hey folks, changing your app icon is now available, for macOS only, in the latest release! 🚀 
See more in our docs 👉  https://docs.warp.dev/appearance/app-icons

No immediate plans to add this feature to Linux or Windows yet, but we'll shout out if this changes 🤞

**Comment by toocomputer** [comment]
This a cool feature @dannyneira, but now custom icons can’t be used at all (icon changes back to the Warp one on app start) 😭

With all due respect (‘cause I really, really, really, love Warp overall), the Warp icon is terrible, and, as other users have pointed out, doesn’t relate in any discernible visual way to a terminal.

I had been using a modified version of [Ghostty’s](https://ghostty.org/) icon, which I really love 👻

My subjective and for-what-it’s-worth opinion on the Warp icon aside, do you think y’all can/will bring back the ability to use custom icons so that the app functions like most other Mac apps in this regard?

**Comment by dannyneira** [comment]
Thanks for the feedback, no immediate plans for that, but I think there are scripts on this thread that help with replacing the icon. We also have a few classic icon options which may help in the meantime if you don't want to go the script route.

**Comment by toocomputer** [comment]
> Afaik, after every update, the icon was replaced. 

@dannyneira: Correct—it was replaced after every update, and I would manually change it back to my custom icon using Finder, but that’s no longer possible. 

I’m not sure it’s possible using a script either, as a script (if my understanding’s correct) simply automates what I was previously doing manually; but now Warp reverts to the Warp icon on every start.

**Comment by dannyneira** [comment]
@toocomputer Please submit a new feature request for the option to apply custom icons. We're generally [good at listening to our user's feedback](https://github.com/warpdotdev/Warp/issues?q=is%3Aissue%20state%3Aclosed%20sort%3Areactions-%2B1-desc), so the more 👍 's the request gets, the higher the likelihood of it becoming a reality. https://github.com/warpdotdev/Warp/issues/new?template=02_feature_request.yml

**Comment by toocomputer** [comment]
Will do! Thanks, @dannyneira! 🎉

(In the meantime, if anyone’s interested, the manual workaround I’ve found to continue using a custom icon is to right-click the app in the applications folder, select Show Package Contents, and delete the `Contents/WarpDockTilePlugin.docktileplugin` directory.)

**Comment by berkus** [comment]
So, the `WarpDockTilePlugin.docktileplugin` provides functionality for custom icons?

**Comment by toocomputer** [comment]
@berkus that seems to be the case from my cursory investigation. not sure how it's working under the hood. i just know that once i deleted it i was able to use my custom icon again.

**Comment by Caffeine19** [comment]
no need at all, the arc browser is totally shit and seems to be dead now, you can only apply the icon it offers, but we are on mac os, we can change any icon we want.
seems the current update, we cant apply our custom icon.
https://github.com/warpdotdev/Warp/issues/5972

and you can just pick some icon from this website https://macosicons.com/#/

**Comment by berkus** [comment]
Lets not make it a post about attacks on my favourite browser, thanks.

**Comment by Caffeine19** [comment]
> Lets not make it a post about attacks on my favourite browser, thanks.

Sorry, I'm maybe too excited, but it's probably the worst browser I've felt

**Comment by dannyneira** [comment]
hey folks, we're tracking the option for custom icons separately here: https://github.com/warpdotdev/Warp/issues/5972

Thanks @toocomputer, for submitting and [providing a workaround](https://github.com/warpdotdev/Warp/issues/5408#issuecomment-2691202147).

If you could please give it a 👍 and subscribe to that request, it would help us gauge interest. We'll post any updates there, so stay tuned!

**Comment by ksylvan** [comment]
> Nice! Do you have plans for Linux support as well?

I asked the Warp agent on my Linux box, and it came up with a solution if you want to try it:

![Image](https://github.com/user-attachments/assets/0648a343-18a6-4abf-ad43-27d981249760)

**Comment by wmorgue** [comment]
For macOS, who trying change icon for app + dock.

Go to directory:

```bash
/Applications/Warp.app/Contents/PlugIns/WarpDockTilePlugin.docktileplugin/Contents/Resources
```

You will see a many icons with `.png` extension.
Just replace `blue.png` to your own icon with same name `blue.png`

<img width="687" alt="Image" src="https://github.com/user-attachments/assets/fa55b925-da56-4f32-bca7-48d04491e339" />

**Comment by toocomputer** [comment]
@wmorgue (+ @dannyneira) that 👆🏾 is indeed a much simpler and more straightforward workaround than what i posted [above](https://github.com/warpdotdev/Warp/issues/5408#issuecomment-2691202147), and indeed what i tried initially.

i must've done something wrong because it didn't work for me then, but it works now.

and unless i'm mistaken, this solution has the added bonus of not having to be repeated every time Warp is updated (which is rather frequent, for better or worse).

**Comment by dannyneira** [comment]
Hey Folks, closing as resolved since there are more icon choices in Settings > Appearance (macOS only)


Please learn more in the docs https://docs.warp.dev/terminal/appearance/app-icons, and let us know if any issues

**Comment by chen-rn** [comment]
hi, are there plans on adding more icon designs?

I think the current set is good but they all look quite similar aside from glassy sky.

would be great to get some icons of completely different design aesthetics. (ie minimalist flat, retro pixel art, cyberpunk neon, skeuomorphic, or nature-inspired themes.)
